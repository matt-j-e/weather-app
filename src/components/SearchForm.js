import React from "react";
import PropTypes from "prop-types";

import "../styles/SearchForm.css";

const SearchForm = ({ searchInput, setSearchInput, onCitySearch }) => {
  const handleInputChange = (event) => setSearchInput(event.target.value);

  return (
    <div className="search-form">
      <form onSubmit={onCitySearch}>
        <input
          onChange={handleInputChange}
          type="text"
          name="city"
          placeholder="City / town"
          value={searchInput}
        />
        <input type="submit" value="Search" />
      </form>
    </div>
  );
};

SearchForm.propTypes = {
  searchInput: PropTypes.string.isRequired,
  setSearchInput: PropTypes.func.isRequired,
  onCitySearch: PropTypes.func.isRequired,
};

export default SearchForm;
