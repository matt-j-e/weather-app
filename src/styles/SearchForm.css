.search-form form {
    text-align: center;
}

/*** MEDIA QUERIES ***/
@media screen and (min-width: 650px) {
    .search-form input {
        font-size: 110%;
    }
}