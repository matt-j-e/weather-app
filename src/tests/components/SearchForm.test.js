import React from "react";
import { render } from "@testing-library/react";
import SearchForm from "../../components/SearchForm";

describe("SearchForm", () => {
  const validProps = {
    searchInput: "Some search string",
    setSearchInput: jest.fn(),
    onCitySearch: jest.fn(),
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <SearchForm
        searchInput={validProps.searchInput}
        setSearchInput={validProps.setSearchInput}
        onCitySearch={validProps.onCitySearch}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct button text", () => {
    const { getByDisplayValue } = render(
      <SearchForm
        searchInput={validProps.searchInput}
        setSearchInput={validProps.setSearchInput}
        onCitySearch={validProps.onCitySearch}
      />
    );
    expect(getByDisplayValue("Search")).toBeInTheDocument();
  });
});
