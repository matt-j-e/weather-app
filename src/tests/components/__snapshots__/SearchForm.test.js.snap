// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm renders correctly 1`] = `
<DocumentFragment>
  <div
    class="search-form"
  >
    <form>
      <input
        name="city"
        placeholder="City / town"
        type="text"
        value="Some search string"
      />
      <input
        type="submit"
        value="Search"
      />
    </form>
  </div>
</DocumentFragment>
`;
